### PR TITLE
Feature/oracle r2dbc connection pool

### PR DESCRIPTION
**Pull request by andCastStatic**
This PR contains a way to enable a connection pool using r2dbc for oracle database. It also has an example on how to implement it in a service.

### CODE REVIEW

File: data/data-samples/sample-data-oracle-reactive-cp-book/pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-runner</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (whoswendy) [inline data/data-samples/sample-data-oracle-reactive-cp-book/pom.xml]: add new line eof please

File: data/synapse-data-oracle/pom.xml [hunk 2]
```diff
@@ -45,10 +45,10 @@
...
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-data-rest</artifactId>
-        </dependency>
+<!--        <dependency>-->
+<!--            <groupId>org.springframework.boot</groupId>-->
+<!--            <artifactId>spring-boot-starter-data-rest</artifactId>-->
+<!--        </dependency>-->
```
Comment (whoswendy) [inline data/synapse-data-oracle/pom.xml]: can we remove commented out code

File: service/service-samples/sample-service-reactive-oracle-cp-book/pom.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (whoswendy) [inline service/service-samples/sample-service-reactive-oracle-cp-book/pom.xml]: add new line eof

File: service/service-samples/sample-service-rest-native-book/pom.xml [hunk 4]
```diff
@@ -28,7 +28,7 @@
         <classifier/>
 
         <!-- Specify main class to allow your native-image-maven-plugin to find it -->
-        <main.class>io.americanexpress.service.book.rest.BookApplication</main.class>
+        <main.class>io.americanexpress.service.book.io.americanexpress.service.book.rest.BookApplication</main.class>
```
Comment (whoswendy) [inline service/service-samples/sample-service-rest-native-book/pom.xml]: undo this change?

File: data/data-samples/sample-data-oracle-reactive-cp-book/src/main/java/io/americanexpress/data/oracle/cp/book/dao/BookRepository.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    Mono<BookEntity> findByAuthor(String author);
+
+    /**
+     * Deletes a BookEntity by title.
+     * @param title The title of the book.
+     * @return VOID
+     */
+    Mono<Void> deleteByTitle(String title);
+
```
Comment (whoswendy) [inline data/data-samples/sample-data-oracle-reactive-cp-book/src/main/java/io/americanexpress/data/oracle/cp/book/dao/BookRepository.java:62]: can we add an example of calling a stored procedure?

Comment (andCastStatic) [reply data/data-samples/sample-data-oracle-reactive-cp-book/src/main/java/io/americanexpress/data/oracle/cp/book/dao/BookRepository.java:62]: This one is just the connection-pool. The one with the stored procedure is a separate branch and it isn't working currently. I haven't taken a look but I can try working on it again.

**Review by whoswendy — APPROVED** [review]

**Review by guttff — APPROVED** [review]
